velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -205,6 +254,126 @@ std::vector<SortingKeyAndOrder> WindowFuzzer::generateSortingKeysAndOrders(
   return results;
 }
 
+template <typename T>
+const std::string& WindowFuzzer::addKRangeOffsetColumnToInputImpl(
+    std::vector<RowVectorPtr>& input,
+    FrameMetadata& frameMetadata,
+    const core::WindowNode::BoundType& frameBoundType,
+    const SortingKeyAndOrder& orderByKey,
+    const std::string& columnName,
+    const std::string& offsetColumnName) {
+  // Use columns as frame bound 50% of time.
+  bool isColumnBound = vectorFuzzer_.coinToss(0.5);
+  auto type = CppToType<T>::create();
+  const auto size = vectorFuzzer_.getOptions().vectorSize;
+  velox::test::VectorMaker vectorMaker{pool_.get()};
+
+  const auto getConstantK = [&](VectorPtr fuzzOffset) -> T {
+    return fuzzOffset->values()->as<T>()[0];
+  };
+  VectorPtr fuzzOffset = vectorFuzzer_.fuzzConstant(type, 1, true);
+  const T constantOffset = getConstantK(fuzzOffset);
+
+  for (auto i = 0; i < FLAGS_num_batches; i++) {
+    BufferPtr offsetColumnValues =
+        AlignedBuffer::allocate<T>(size, pool_.get());
+    auto* offsetColumnValuesPtr = offsetColumnValues->asMutable<T>();
+    BufferPtr offsetColumnNulls = allocateNulls(size, pool_.get());
+    auto* offsetColumnRawNulls = offsetColumnNulls->asMutable<uint64_t>();
+
+    BufferPtr columnValues = AlignedBuffer::allocate<T>(size, pool_.get());
+    BufferPtr columnNulls = allocateNulls(size, pool_.get());
+    auto* columnValuesPtr = columnValues->asMutable<T>();
+    auto* columnRawNulls = columnNulls->asMutable<uint64_t>();
+
+    auto orderByCol = input[i]->childAt(orderByKey.key_);
+    const SelectivityVector allRows(size);
+    DecodedVector decodedVector(*orderByCol, allRows);
+
+    for (auto j = 0; j < size; j++) {
+      if (decodedVector.isNullAt(j)) {
+        if (isColumnBound) {
+          bits::setNull(columnRawNulls, j, true);
+        }
+        bits::setNull(offsetColumnRawNulls, j, true);
+      } else {
+        T offsetValue;
+        if (isColumnBound) {
+          offsetValue = getConstantK(vectorFuzzer_.fuzzConstant(type, 1, true));
```

Comment:
Looks like this is generated in exactly the same way as `constantOffset`. Can we use constantOffset directly then?